SpatialGDK/Source/SpatialGDK/Private/Tests/CrossServerRPCHandlerTest.cpp

```diff
@@ -0,0 +1,270 @@
+	EntityComponentOpListBuilder Builder;
+	Builder.AddEntityCommandRequest(TestEntityId, QueueingRequestId, CreateCrossServerCommandRequest());
+	OpLists.Add(MoveTemp(Builder).CreateOpList());
+	ListsOfOpLists.Add(MoveTemp(OpLists));
+
+	OpLists = TArray<OpList>();
+	Builder = EntityComponentOpListBuilder();
+	Builder.AddEntityCommandRequest(TestEntityId, QueueingRequestId, CreateCrossServerCommandRequest());
+	OpLists.Add(MoveTemp(Builder).CreateOpList());
+
+	ListsOfOpLists.Add(MoveTemp(OpLists));
+	ConnHandler->SetListsOfOpLists(MoveTemp(ListsOfOpLists));
+	ViewCoordinator Coordinator(MoveTemp(ConnHandler), nullptr, ComponentSetData);
+	CrossServerRPCHandler Handler(Coordinator, MakeUnique<MockRPCExecutor>());
+	Coordinator.Advance(0.f);
+	Handler.ProcessMessages(Coordinator.GetViewDelta().GetWorkerMessages());
+
+	Coordinator.Advance(0.f);
+	Handler.ProcessMessages(Coordinator.GetViewDelta().GetWorkerMessages());
+	const auto& QueuedRPCs = Handler.GetQueuedCrossServerRPCs();
+	TestEqual("Number of queued up Cross Server RPCs", QueuedRPCs.Num(), 1);
+	if (!QueuedRPCs.Contains(TestEntityId))
+	{
+		TestTrue("TestEntityId not in queued up RPCs", false);
+	}
+	else
+	{
+		TestEqual("Number of queued up Cross Server RPCs", QueuedRPCs[TestEntityId].Num(), 1);
+	}
+
+	return true;
+}
+
+CROSSSERVERRPCHANDLER_TEST(GIVEN_rpc_WHEN_resolved_and_queue_THEN_queue)
+{
+	AddExpectedError(QueueingCommand, EAutomationExpectedErrorFlags::Exact, 3);
+
+	TArray<TArray<OpList>> ListsOfOpLists;
+	TUniquePtr<MockConnectionHandler> ConnHandler = MakeUnique<MockConnectionHandler>();
+
+	TArray<OpList> OpLists;
+	EntityComponentOpListBuilder Builder;
+	Builder.AddEntityCommandRequest(TestEntityId, QueueingRequestId, CreateCrossServerCommandRequest());
+	OpLists.Add(MoveTemp(Builder).CreateOpList());
+	ListsOfOpLists.Add(MoveTemp(OpLists));
+
+	Builder = EntityComponentOpListBuilder();
+	Builder.AddEntityCommandRequest(TestEntityId, SuccessRequestId, CreateCrossServerCommandRequest());
+	OpLists = TArray<OpList>();
+	OpLists.Add(MoveTemp(Builder).CreateOpList());
+
+	ListsOfOpLists.Add(MoveTemp(OpLists));
+	ConnHandler->SetListsOfOpLists(MoveTemp(ListsOfOpLists));
+	ViewCoordinator Coordinator((MoveTemp(ConnHandler)), nullptr, ComponentSetData);
+	CrossServerRPCHandler Handler(Coordinator, MakeUnique<MockRPCExecutor>());
+	Coordinator.Advance(0.f);
+	Handler.ProcessMessages(Coordinator.GetViewDelta().GetWorkerMessages());
+
+	Coordinator.Advance(0.f);
+	Handler.ProcessMessages(Coordinator.GetViewDelta().GetWorkerMessages());
+	const auto& QueuedRPCs = Handler.GetQueuedCrossServerRPCs();
+	TestEqual("Number of queued up Cross Server RPCs", QueuedRPCs.Num(), 1);
+	if (!QueuedRPCs.Contains(TestEntityId))
+	{
+		TestTrue("TestEntityId not in queued up RPCs", false);
+	}
+	else
+	{
+		TestEqual("Number of queued up Cross Server RPCs", QueuedRPCs[TestEntityId].Num(), 2);
+	}
+	return true;
+}
+
+CROSSSERVERRPCHANDLER_TEST(GIVEN_rpc_WHEN_unresolved_THEN_queue)
```

Comment:
Where is the "unresolved" part in this test ? I cannot figure out what's different from the previous ones.

Reply:
this is the test to check whether the RPC gets queued correctly. in the tests this is gated based on the request id they have as  simulating the unresolved/resolved part would be tricky